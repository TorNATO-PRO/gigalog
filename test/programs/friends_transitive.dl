friend(alice, bob).
friend(bob, carol).
friend(carol, dave).

connected(X, Y) :- friend(X, Y).
connected(X, Y) :- friend(X, Z), connected(Z, Y).
