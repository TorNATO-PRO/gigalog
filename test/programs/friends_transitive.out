connected(alice,bob)
connected(alice,carol)
connected(alice,dave)
connected(bob,carol)
connected(bob,dave)
connected(carol,dave)
friend(alice,bob)
friend(bob,carol)
friend(carol,dave)
